flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/CalcITCase.scala

```diff
@@ -433,4 +435,61 @@ class CalcITCase extends StreamingTestBase {
     val expected = List("{a=0.12, b=0.50}")
     assertEquals(expected.sorted, sink.getAppendResults.sorted)
   }
+
+  @Test
+  def testCurrentWatermark(): Unit = {
```

Comment:
test the example that we also documented using WHERE

Reply:
test some more complex example, maybe with a window or window join before calling `CURRENT_WATERMARK`

Reply:
In the SQL client I now tried a bunch of different things like windowing, interval joins etc. and they all seemed to work. Of course I'm using simple, non-realistic setups. Let me know if there's anything specific to test? You have a better idea of in which scenarios this can happen. :-)